### PR TITLE
Running fix

### PR DESCRIPTION
**Pull request by cbosdo**
### What does this PR do?

Report an error when a VM fails to start while applying `virt.running` state

### What issues does this PR fix or reference?

issue #47972 

### Previous Behavior

When applying `virt.running` state to a VM that fails to work (like network not started) the return message was that the domain was already up and running

### New Behavior

When applying `virt.running` state to a VM that fails to work (like network not started) the return message will be the libvirt error and the result will be set to `False`.

### Tests written?

Yes, added virt.running test cases

### Commits signed with GPG?

Yes

### CODE REVIEW

File: tests/unit/states/test_libvirt.py [hunk 1]
```diff
@@ -23,13 +25,30 @@
...
+class LibvirtMock(MagicMock):  # pylint: disable=too-many-ancestors
+    '''
+    libvirt library mockup
+    '''
+
+    class libvirtError(Exception):  # pylint: disable=invalid-name
+        '''
+        libvirt error mockup
+        '''
+        pass
```
Comment (isbm) [inline tests/unit/states/test_libvirt.py]: No need this after you have a docstring. :wink:

**Review by isbm — CHANGES_REQUESTED** [review]

File: tests/unit/states/test_libvirt.py (hunk 1 above)
Comment (cbosdo) [reply tests/unit/states/test_libvirt.py]: Oh ok, I'll remove it then

**Comment by cbosdo** [comment]
@isbm Fixed, thanks for the review

**Review by isbm — APPROVED** [review]

**Comment by cbosdo** [comment]
@gtmanfred , @rallytime Could you have a look at this PR?

**Review by gtmanfred — APPROVED** [review]

File: salt/states/virt.py [hunk 2]
```diff
@@ -247,13 +247,17 @@ def running(name, **kwargs):
...
         except CommandExecutionError:
             kwargs = salt.utils.args.clean_kwargs(**kwargs)
             __salt__['virt.init'](name, cpu=cpu, mem=mem, image=image, **kwargs)
             ret['changes'][name] = 'Domain defined and started'
             ret['comment'] = 'Domain {0} defined and started'.format(name)
-    except libvirt.libvirtError:
-        ret['comment'] = 'Domain {0} exists and is running'.format(name)
+    except libvirt.libvirtError as err:
+        # Something bad happened when starting the VM, report it
+        ret['comment'] = str(err)
```
Comment (rallytime) [inline salt/states/virt.py]: I think this should be `six.text_type` for Py2/Py3 compatibility.

**Review by rallytime — CHANGES_REQUESTED** [review]
One comment from me.

**Comment by cbosdo** [comment]
@rallytime fixed

**Review by rallytime — APPROVED** [review]

**Pull request merged**